internal/pipeline/pipeline_test.go

```diff
@@ -66,138 +31,107 @@ func checkOutput(t *testing.T, tc testcase, actual *Pipeline, err error) {
 	} else if !assert.NoError(t, err, "error is not expected.") {
 		t.FailNow()
 	}
-	if !assert.True(t, isPipelineEqual(t, tc.Expected, *actual),
+	if !assert.True(t, isPipelineEqual(t, *tc.Expected, *actual),
 		"pipelines don't equal.") {
 		t.FailNow()
 	}
 }
 
-func TestFromReader(t *testing.T) {
-	for name, tc := range testcases {
-		tc := tc
-		name := name
-		t.Run(name, func(t *testing.T) {
-			r := bytes.NewBufferString(tc.Input)
-			actual, err := FromReader(r)
-			checkOutput(t, tc, actual, err)
-		})
-	}
-}
+func TestPipelineValidate(t *testing.T) {
```

Comment:
I feel like these tests can be moved to where Pipeline type exists. Also, reading the pipeline as part of Kptfile will be better because that will be our regular usage.